ddpui/api/webhook_api.py

```
@@ -77,3 +80,44 @@
+                "session_id": llm_session.session_id,
+                "created_at": llm_session.created_at,
+                "user_prompts": llm_session.user_prompts,
+                "assistant_prompt": llm_session.assistant_prompt,
+                "response": llm_session.response,
+                "session_status": llm_session.session_status,
+            },
+        }
+
+    except Exception as err:
+        logger.error(f"Error retrieving failure summary for flow run {flow_run_id}: {str(err)}")
+        raise HttpError(500, f"Failed to retrieve failure summary: {str(err)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)